pulsar-broker/src/test/java/org/apache/pulsar/client/api/AuthorizationProducerConsumerTest.java

```diff
@@ -538,6 +538,60 @@ public void testAuthData() throws Exception {
         log.info("-- Exiting {} test --", methodName);
     }
 
+    @Test
+    public void testPermissionForProducerCreateInitialSubscription() throws Exception {
+        log.info("-- Starting {} test --", methodName);
+
+        conf.setAuthorizationProvider(PulsarAuthorizationProvider.class.getName());
+        setup();
+
+        String lookupUrl = pulsar.getBrokerServiceUrl();
+
+        final String invalidRole = "invalid-role";
+        final String producerRole = "producer-role";
+        final String topic = "persistent://my-property/my-ns/my-topic";
+        final String initialSubscriptionName = "init-sub";
+        Authentication adminAuthentication = new ClientAuthentication("superUser");
+        Authentication authenticationInvalidRole = new ClientAuthentication(invalidRole);
+        Authentication authenticationProducerRole = new ClientAuthentication(producerRole);
+        @Cleanup
+        PulsarAdmin admin =
+                PulsarAdmin.builder().serviceHttpUrl(brokerUrl.toString()).authentication(adminAuthentication).build();
+
+        admin.clusters().createCluster("test", ClusterData.builder().serviceUrl(brokerUrl.toString()).build());
+        admin.tenants().createTenant("my-property",
+                new TenantInfoImpl(Sets.newHashSet("appid1", "appid2"), Sets.newHashSet("test")));
+        admin.namespaces().createNamespace("my-property/my-ns", Sets.newHashSet("test"));
+        admin.topics().grantPermission(topic, invalidRole, Collections.singleton(AuthAction.produce));
+        admin.topics().grantPermission(topic, producerRole, Sets.newHashSet(AuthAction.produce, AuthAction.consume));
```

Comment:
Should check if the client does not have the consume permission, the producer should fail.